src/include/replication/walsender.h

```diff
@@ -36,6 +36,9 @@ extern PGDLLIMPORT int max_wal_senders;
 extern PGDLLIMPORT int wal_sender_timeout;
 extern PGDLLIMPORT bool log_replication_commands;
 
+struct XLogReaderRoutine;
```

Comment:
Remove the forward declaration. Postgres is full of headers that don't actually compile by themselves. Really annoying.